### PR TITLE
Feature/ss4 upgrade

### PR DESCRIPTION
**Pull request by phalkunz**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
     "description": "Flowchart silverstripe module.",
-    "type": "silverstripe-module",
+    "type": "silverstripe-vendormodule",
     "keywords": ["silverstripe"],
-    "minimum-stability": "dev",
     "license": "BSD-3-Clause",
     "require": {
-        "silverstripe/cms": "~3.1",
-        "silverstripe/framework": "~3.1"
+        "silverstripe/cms": "^4",
```
Comment (pjayme) [inline composer.json:8]: Does it matter if this isn't the latest version of SS4? I.e. should it be `"^4.2"` ?

File: src/admin/FlowchartAdmin.php [hunk 2]
```diff
@@ -30,16 +51,25 @@ public function providePermissions()
         );
     }
 
+    /**
+     * @inheritdoc
```
Comment (pjayme) [inline src/admin/FlowchartAdmin.php:55]: Not too sure if this should say `@inheritdoc`? Same with others below.

File: src/model/Flowchart.php [hunk 3]
```diff
@@ -122,10 +169,10 @@ public static function shortcodeHandler($arguments, $content = null, $parser = n
             return null;
         }
 
-        Requirements::css('silverstripe-flowchart/css/flowchart.css');
+        Requirements::css('chtombleson/silverstripe-flowchart:css/flowchart.css');
 
-        Requirements::javascript(FRAMEWORK_DIR . '/thirdparty/jquery/jquery.min.js');
-        Requirements::javascript('silverstripe-flowchart/javascript/flowchart.js');
+        // Requirements::javascript(FRAMEWORK_DIR . '/thirdparty/jquery/jquery.min.js');
```
Comment (pjayme) [inline src/model/Flowchart.php:174]: Is this line meant to be removed?

File: src/model/FlowchartFeedback.php [hunk 4]
```diff
@@ -1,36 +1,61 @@
...
-        'Feedback' => 'Text'
+        'Feedback' => 'Text',
     ];
 
+    /**
+     * @var array
+     */
     private static $has_one = [
-        'Flowchart' => 'Flowchart'
+        'Flowchart' => Flowchart::class,
```
Comment (pjayme) [inline src/model/FlowchartFeedback.php:21]: `Flowchart` appears to be missing namespace

File: composer.json (hunk 1 above)
Comment (phalkunz) [reply composer.json:8]: It means the latest of `4.x`

File: src/model/Flowchart.php (hunk 3 above)
Comment (phalkunz) [reply src/model/Flowchart.php:174]: Good catch.

**Pull request merged**